legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -92,121 +96,88 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
+                if (atomicTestIds.contains(testedError._id()) && results.stream().noneMatch(t -> t.atomicTestId.equals(testedError._id())))
                 {
-                    org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult testResult = executeMappingDataTest(dataTest, context);
-                    testResult.testable = getElementFullPath(pureMapping, pureModel.getExecutionSupport());
-                    testResult.testSuiteId = mappingFunctionTestSuite.id;
-                    results.add(testResult);
+                    TestError testError = new TestError();
+                    testError.atomicTestId = testedError._id();
+                    testError.error = e.toString();
+                    results.add(testError);
                 }
             }
         }
-        catch (Exception e)
-        {
-            throw new RuntimeException("Exception occurred executing service test suites.\n" + e);
-        }
         return results;
     }
 
-    public Root_meta_pure_runtime_Runtime_Impl buildRuntimeForMultiInputTestSuite(List<MappingDataTest> dataTests,  MappingTestRunnerContext context)
-    {
-        Root_meta_pure_runtime_Runtime_Impl runtime = new Root_meta_pure_runtime_Runtime_Impl("");
-        if (dataTests.size() > 0)
-        {
-            MappingDataTest mappingDataTest = dataTests.get(0);
-            List<Pair<String, EmbeddedData>> connectionInfo = mappingDataTest.storeTestData.stream().map(testData -> Tuples.pair(testData.store, EmbeddedDataHelper.resolveEmbeddedData(context.getPureModelContextData(), testData.data))).collect(Collectors.toList());
-            List<Pair<Connection, List<Closeable>>> connections = connectionInfo.stream()
-                    .map(pair -> this.factories.collect(f -> f.tryBuildTestConnectionsForStoreWithMultiInputs(context.dataElements, resolveStore(context.getPureModelContextData(), pair.getOne()), pair.getTwo())).select(Objects::nonNull).select(Optional::isPresent)
-                    .collect(Optional::get).getFirstOptional().orElseThrow(() -> new UnsupportedOperationException("Unsupported store type for:'" + pair.getOne() + "' mentioned while running the mapping tests"))).collect(Collectors.toList());
-            connections.stream().forEach(conn -> runtime._connectionsAdd(conn.getOne().accept(context.connectionVisitor)));
-        }
-        return runtime;
-
-    }
-
-    private TestResult executeMappingDataTest(MappingDataTest dataTest, MappingTestRunnerContext context)
+    private TestResult executeMappingTest(MappingTest mappingTest,  MappingTestRunnerContext context)
     {
         List<Pair<Connection, List<Closeable>>> connections = Lists.mutable.empty();
         try
         {
-            Optional<InputStream> inputStream = buildInputStreamIfPossible(dataTest, context);
-            if (!inputStream.isPresent())
+            Assert.assertTrue(mappingTest.assertions.size() == 1, () -> "Mapping Test expected to have one assertion");
```

Comment:
We already have this in compilation phase, any reason we have it again. Seems we are leaking compilation concerns to execution